core/iwasm/interpreter/wasm_loader.c

```diff
@@ -2255,9 +2255,16 @@ load_type_section(const uint8 *buf, const uint8 *buf_end, WASMModule *module,
 static void
 adjust_table_max_size(uint32 init_size, uint32 max_size_flag, uint32 *max_size)
 {
-    uint32 default_max_size = init_size * 2 > WASM_TABLE_MAX_SIZE
-                                  ? init_size * 2
-                                  : WASM_TABLE_MAX_SIZE;
+    uint32 default_max_size;
+
+    bool overflow = __builtin_umul_overflow(init_size, 2, &default_max_size);
+    if (overflow) {
+        default_max_size = UINT_MAX;
+    }
+    else {
+        if (default_max_size < WASM_TABLE_MAX_SIZE)
+            default_max_size = WASM_TABLE_MAX_SIZE;
+    }
```

Comment:
It is not easy to ensure that __builtin_umul_overflow is supported in all platforms and targets, how about changing to be somewhat like:
```C
    if (init_size * 2 < init_size) {
        default_max_size = UINT32_MAX;
    }
    else {
        default_max_size = init_size * 2 > WASM_TABLE_MAX_SIZE
                                  ? init_size * 2
                                  : WASM_TABLE_MAX_SIZE;
    }
```
And had better use UINT32_MAX instead of UINT_MAX?